### PR TITLE
Date Filters: Increase min-width to account for visible scroll bars, improve expand-collapse behaviour [GAUD-6589]

### PR DESCRIPTION
**Pull request by margaree**
[Jira ticket](https://desire2learn.atlassian.net/jira/software/c/projects/GAUD/boards/330?selectedIssue=GAUD-6589)

This covers a few things:
- When scroll bars are visible not just on scroll the default min-width for the date filter list items is a bit too small (can trigger wrapping), so this increases it
- Jitter in expand/collapse behaviour:
    - Cause 1: `input-date-time-range-to` component and all its margins
    - Cause 2: `overflow-y` scroll temporarily being triggered briefly during the expand-collapse animation (does not happen when reduce motion)

### CODE REVIEW

File: components/filter/filter.js [hunk 1]
```diff
@@ -841,6 +847,15 @@ class Filter extends FocusMixin(LocalizeCoreElement(RtlMixin(LitElement))) {
...
 	}
 
+	async _handleExpandCollapse(e) {
+		const eventPromise = e.target.expanded ? e.detail.expandComplete : e.detail.collapseComplete;
+		const parentListItem = e.target.closest('d2l-list-item');
+		parentListItem.classList.add('expanding-content');
+
+		await eventPromise;
+		parentListItem.classList.remove('expanding-content');
+	}
```
Comment (margaree) [inline components/filter/filter.js:857]: This seemed a bit weird so let me know if there's an alternative I'm missing. I tried `overflow-y: hidden` elsewhere and the list item was the only spot where it mattered.

File: components/inputs/input-date-time-range-to.js [hunk 2]
```diff
@@ -41,65 +41,49 @@ class InputDateTimeRangeTo extends SkeletonMixin(LocalizeCoreElement(LitElement)
...
+			.d2l-input-date-time-range-to-container-block .d2l-input-date-time-range-end-container {
+				display: block;
+				margin-bottom: 1.2rem;
 			}
-			.d2l-input-date-time-range-to-to {
+			.d2l-input-date-time-range-to-container-block .d2l-input-date-time-range-to-to {
 				display: inline-block;
 				margin-bottom: 0.6rem;
-				margin-right: 0.9rem;
+				margin-inline-end: 0.9rem;
```
Comment (margaree) [inline components/inputs/input-date-time-range-to.js:83]: I actually don't think this would be needed.

File: components/inputs/input-date-time-range-to.js
Comment (margaree) [inline components/inputs/input-date-time-range-to.js:1]: What I did here:
- created flex case and block case sections, copied the styles to both (flex = not wrapped, block = wrapped; we want the "to" on a separate line when wrapped)
- simplified each case to only contain the styles that mattered to it
- removed margins from the flex case, replaced with `column-gap`
- removed the "to" from the end container
- removed `dir="rtl"` usages from the css

It's totally possible there's a better way to do this. In testing this seems fine, but this change is definitely relying on the visual diffs. I did find that some of the visual diffs (notably ones with error tooltips) needed a bit more time to get the correct height.

File: components/inputs/test/input-date-time-range.vdiff.js
Comment (dbatiste) [inline components/inputs/test/input-date-time-range.vdiff.js:1]: Nit: could find a way to reduce some of the duplication here.

**Review by dbatiste — APPROVED** [review]

**Pull request merged**